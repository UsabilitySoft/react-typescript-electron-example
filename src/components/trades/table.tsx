import React, { useState, useEffect } from 'react';
import { Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';

import { Fields } from '../../helpers/language';
import { ITradeRow, TradeRow } from './table.row';

interface ITradesTableProps {
  tradesData: ITradeRow[];
}

export const TradesTable: React.FC<ITradesTableProps> = (props) => {

  return (
    <Table padding="default">
      <TableHead>
        <TableRow>
          <TableCell>{Fields.PRODUCT_NAME.TITLE}</TableCell>
          <TableCell>{Fields.BOOK_NAME.TITLE}</TableCell>
          <TableCell>{Fields.CANCELLED.TITLE}</TableCell>
          <TableCell>{Fields.END_DATE.TITLE}</TableCell>
          <TableCell>{Fields.MATCHED_TRADE.TITLE}</TableCell>
          <TableCell>{Fields.SIDE.TITLE}</TableCell>
          <TableCell>{Fields.START_DATE.TITLE}</TableCell>
          <TableCell>{Fields.TIME_CREATED.TITLE}</TableCell>
          <TableCell>{Fields.TRADE_DATE.TITLE}</TableCell>
          <TableCell>{Fields.TRADE_DISPLAY_VOLUME.TITLE}</TableCell>
          <TableCell>{Fields.TRADE_PRICE.TITLE}</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        { props.tradesData.map((row: ITradeRow) => {
            return (
              <TradeRow id={row.id} row={row}></TradeRow>
            )
          }) 
        }
      </TableBody>
    </Table>
  );
};