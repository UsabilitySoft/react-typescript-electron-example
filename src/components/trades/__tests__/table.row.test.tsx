import React from 'react';
import renderer from 'react-test-renderer';

import { mockRow } from '../../../helpers/sample.data';
import { TradeRow } from '../table.row';

it('renders correctly when there is one row', () => {
  const tree = renderer.create(<TradeRow row={mockRow} />).toJSON();
  expect(tree).toMatchSnapshot();
});