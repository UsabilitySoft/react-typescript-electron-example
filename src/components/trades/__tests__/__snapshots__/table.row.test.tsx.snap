// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is one row 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    BUY
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    2019-09-06
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    3% GC/3.5% Barges Rot
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    Fuel
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
    style={null}
  >
    22.50
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    6
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  />
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  />
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    2020-12-01
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    2020-12-31
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    2019-09-06T17:39:16.477417Z
  </td>
</tr>
`;
